### PR TITLE
refactored code

### PR DESCRIPTION
**Pull request by AdityaRaimec22**

### CODE REVIEW

File: src/controllers/communityController.js [hunk 1]
```diff
@@ -93,35 +94,43 @@ const deleteCommunity = async (req, res) => {
...
-            } else if (results.rows.length == 0) {
+const getAllTripsOfCommunity = async (req, res, next) => {
+    // Collect unique dates, origins, and destinations
+    const tripIds = new Set();
+    const community_id = req.params.community_id;
+    
+    try {
+
+        let sqlQuery = `
+        SELECT DISTINCT CT.trip_id
```
Comment (mviswanathsai) [inline src/controllers/communityController.js]: Here, we are getting the trip ids in one query, then passing it to the queryTrips. Which once again performs a query to get all trips. After that you filter those trips against these trip ids. 
instead we can simply construct this one query to achieve the parameters. 
The existing query directly fetches all the trip rows in the community, we can change the query params with minimal effort.
Also, I do not understand why you need to fetch all users to ultimately only consider the trips relevant to the user, while you directly have the trips revelant to the user from this query.

**Review by mviswanathsai — CHANGES_REQUESTED** [review]

File: src/controllers/communityController.js (hunk 1 above)
Comment (AdityaRaimec22) [reply src/controllers/communityController.js]: Ah! I saw that now I did a mistake actually I have to use the sent tripIds not to again perform a query to get all the trips in the queryTrips function I will fix it.

Comment (AdityaRaimec22) [reply src/controllers/communityController.js]: Actually what is happening we are getting all the tripIds relevant to a community from the getAllTripsOfCommunity function then we are sending these tripIds to the queryTrips function where now this function is getting all the information of the tripIds and then different filters are applied.

Comment (mviswanathsai) [reply src/controllers/communityController.js]: lets leave the filters for later, for now do not change this function.

Comment (mviswanathsai) [reply src/controllers/communityController.js]: Revert the changes in that function and make a commit, I will merge it

**Review by mviswanathsai — APPROVED** [review]

**Pull request merged**